### PR TITLE
[process-agent] generate unique uid for static pods

### PR DESCRIPTION
**Pull request by nammn**
### What does this PR do?

- generating a unique hash based on `host, podName, namespace, clusterName` for staticPods as `uid`
    - related PR on k8 https://github.com/kubernetes/kubernetes/pull/87461
- `invoke deps` on go version `go version go1.14.3 darwin/amd64`
- `invoke generate-licenses`

### Motivation

- Jira: CAP-108
Static Pod Id's are actually not unique across nodes. Because "only" the pod config get's hashed. In case if this pod config is used again on another node the uid will be the same. 
The proposed solution is generating a unique hash based on `host, podName, namespace, clusterName`.

### Additional Notes

The `DCA` would not "see" the staticpod, because it basically does a kind of `k get pods --field-selector=spec.nodeName="" --all-namespaces`
Looking only for pods which does not have a node scheduled.
But as the static pod is directly put on a kubelet, `nodeName` should be set. 
I tested it with a pod which is not schedulable (too high request) was not shown with `k get pods` but `curl kubelet/pods` has shown it as `failed` and `nodeName` set.

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: pkg/process/util/orchestrator/pod.go [hunk 1]
```diff
@@ -22,6 +23,8 @@ import (
 	jsoniter "github.com/json-iterator/go"
 	yaml "gopkg.in/yaml.v2"
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
```
Comment (nammn) [inline pkg/process/util/orchestrator/pod.go:26]: the license check fails because of those imports..

**Review by mx-psi — APPROVED** [review]

File: pkg/process/util/orchestrator/pod_test.go [hunk 2]
```diff
@@ -727,3 +727,15 @@ func TestChunkPods(t *testing.T) {
...
 }
+
+func TestGenerateUniqueStaticPodHash(t *testing.T) {
+	hostName := "agent-dev-tim"
+	podName := "nginxP"
+	namespace := "kube-system"
+	clusterName := "something"
+
+	uniqueHash := generateUniqueStaticPodHash(hostName, podName, namespace, clusterName)
+	uniqueHashAgain := generateUniqueStaticPodHash(hostName, podName, namespace, clusterName)
```
Comment (mfpierre) [inline pkg/process/util/orchestrator/pod_test.go:738]: could we also add the hardcoded version of it in the test just to see what to expect?

File: releasenotes/notes/fix-staticpod-uid-0348d0703c0d6961.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Fix uid of static pods to be unique across nodes and cluster.
```
Comment (mfpierre) [inline releasenotes/notes/fix-staticpod-uid-0348d0703c0d6961.yaml:11]: not sure it's worth adding release notes for now, cc @hkaj

Comment (hkaj) [reply releasenotes/notes/fix-staticpod-uid-0348d0703c0d6961.yaml:11]: yeah no need for now, we'll see once more information about the feature is available publicly :)

**Review by mfpierre — APPROVED** [review]
Code changes look good to me 👍 
still a bit unsure about the licences changes 🤔

File: pkg/process/util/orchestrator/pod.go [hunk 4]
```diff
@@ -366,3 +375,15 @@ func GetConditionMessage(p *v1.Pod) string {
 	}
 	return ""
 }
+
+// this should generate a unique id because:
+// postName + namespace = unique per host
```
Comment (xlucas) [inline pkg/process/util/orchestrator/pod.go:380]:
```suggestion
// podName + namespace = unique per host
```

File: pkg/process/util/orchestrator/pod.go [hunk 5]
```diff
@@ -366,3 +375,15 @@ func GetConditionMessage(p *v1.Pod) string {
...
+// postName + namespace + host + clustername = unique
```
Comment (xlucas) [inline pkg/process/util/orchestrator/pod.go:381]:
```suggestion
// podName + namespace + host + clustername = unique
```

**Pull request merged**